PureStoragePlugin.pm

```
@@ -1097,19 +1158,29 @@ sub activate_volume {
   my ( $class, $storeid, $scfg, $volname, $snapname, $cache ) = @_;
   print "Debug :: PVE::Storage::Custom::PureStoragePlugin::sub::activate_volume\n" if $DEBUG;
 
-  $class->purestorage_volume_connection( $scfg, $volname, 1 );
-
-  $class->map_volume( $storeid, $scfg, $volname, $snapname );
-  return 1;
+  my $lun = $class->purestorage_volume_connection( $scfg, $volname );
+  if (!$lun) {
+    $class->purestorage_volume_connection( $scfg, $volname, 1 );
+  }
+  my $protocol = $scfg->{ protocol } // $default_protocol;
```

Again, I see no real need to check for existing connection if in the end we just want to make sure the volume is connected.

---

Here it is okay to call POST and DELETE unconditally and ignore the errors if you want to spare the api call.